site-src/geps/gep-713.md

```diff
@@ -0,0 +1,409 @@
+a given namespace can attach it to any resource within that namespace.
+
+![Simple Ingress Example](images/713-ingress-simple.png)
+
+Attaching policy to Gateway resources for ingress use cases is relatively
+straightforward. A policy can reference the resource it wants to apply to.
+Access is granted with RBAC - anyone that has access to create a RetryPolicy in
+a given namespace can attach it to any resource within that namespace.
+
+![Complex Ingress Example](images/713-ingress-complex.png)
+
+### Policy Attachment for Mesh
+Although there is a great deal of overlap between ingress and mesh use cases,
+mesh enables more complex policy attachment scenarios. For example, you may want
+to apply policy to requests from a specific namespace to a backend in another
+namespace. 
+
+![Simple Mesh Example](images/713-mesh-simple.png)
+
+Policy attachment can be quite simple with mesh. Policy can be applied to any
+resource in any namespace but it can only apply to requests from the same
+namespace if the target is in a different namespace.
+
+At the other extreme, policy can be used to apply to requests from a specific
+workload to a backend in another namespace. A route can be used to intercept
+these requests and split them between different backends (foo-a and foo-b in
+this case).
+
+![Complex Mesh Example](images/713-mesh-complex.png)
+
+### Policy TargetRef API
+
+Each Policy resource should include a single `targetRef` field. It must not
+target more than one resource at a time, but it can be used target larger
+resources such as Gateways or Namespaces that may apply to multiple child
+resources. This field should have the following structure:
+
+```go
+// PolicyTargetReference identifies an API object to apply policy to.
+type PolicyTargetReference struct {
+    // Group is the group of the target resource.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    Group string `json:"group"`
+
+    // Kind is kind of the target resource.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    Kind string `json:"kind"`
+
+    // Name is the name of the target resource.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    Name string `json:"name"`
+
+    // SectionName is the name of a section within the target resource. When 
+    // unspecified, this targets the entire resource. In the following
+    // resources, SectionName is interpreted as the following:
+    // * Gateway: Listener Name
+    // * Route: Rule Name
+    // * Service: Port Name
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    // +optional
+    SectionName string `json:"sectionName,omitempty"`
```

Comment:
I'd really hate to have multiple levels of hierarchy here, though I admit it is possible. If we really have to, we could consider adding a `SectionType` or `SubsectionName` field.

Reply:
Does the "SectionName" hierarchy provides scalability like  
"Namespace.SectionName1.SectionName2.svc...."

or just one more level "Namespace.SectionName.svc..."

Reply:
Question - do we really need sectionName for v1alpha2? Are there any core/MVP use-cases that need it right now? It feels like we're a little less sure how this should work and because it would be a pretty simple additive change, we could push this off for later to first get requests for it from users to understand the use-cases before we decide to design it.